### ISSUE TITLE
cube(dim, up, low)

### ISSUE THREAD

**Issue by kalmarek** [oscar-system/Polymake.jl#129]
doesn't seem to work:

```julia
p = Polytope.cube(3, 1//4, -1//4)
p.VERTICES
pm::Matrix<pm::Rational>
1 -1 -1 -1
1 1 -1 -1
1 -1 1 -1
1 1 1 -1
1 -1 -1 1
1 1 -1 1
1 -1 1 1
1 1 1 1
```

**Comment by kalmarek** [comment]
but the problem is on the bottom:
```
a = Polymake.internal_call_function("polytope::cube", String[], Any[3, 1//4, -1//4])
Polymake.convert_from_property_value(a).VERTICES

pm::Matrix<pm::Rational>
1 -1 -1 -1
1 1 -1 -1
1 -1 1 -1
1 1 1 -1
1 -1 -1 1
1 1 -1 1
1 -1 1 1
1 1 1 1
```
@sebasguts mind looking into this?
```

**Comment by sebasguts** [comment]
@kalmarek The reason is that we do not convert `Rational{Whatever}` when calling the function. For example, one can do

```
julia> a = Polymake.internal_call_function("polytope::cube", String[], Any[3, Polymake.pm_Rational(1//4), Polymake.pm_Rational(-1//4)])
Polymake::polytope::Polytope__Rational=ARRAY(0x9c71cb8)

julia> Polymake.convert_from_property_value(a).VERTICES
pm::Matrix<pm::Rational>
1 -1/4 -1/4 -1/4
1 1/4 -1/4 -1/4
1 -1/4 1/4 -1/4
1 1/4 1/4 -1/4
1 -1/4 -1/4 1/4
1 1/4 -1/4 1/4
1 -1/4 1/4 1/4
1 1/4 1/4 1/4
```

Should we do `BigInt` and `Rational{BigInt}` and `Rational{Int64}`, too?

**Comment by kalmarek** [comment]
where this conversion should happen?
at the moment we don't seem to do anything to args, but `c_arguments`;

I'm reworking conversion stuff at the moment (and extending rationals on the way)

**Comment by sebasguts** [comment]
If you want to have the conversion even for the low-level caller, you need to do it in `polymake_caller.cpp`. 

This one does not take `args`, but needs to store them in an `pm::OptionSet` beforehand, so the conversion needs to happen before creating said `OptionSet`.

If you want to do conversion for `args`, you need to do it in the high-level caller or the option set constructor. Does that makes sense?

**Comment by kalmarek** [comment]
fixed by #133
